src/MiddlewareListener.php

```diff
@@ -142,7 +143,10 @@ private function createPipeFromSpec(
             if (is_string($middlewareToBePiped) && $serviceLocator->has($middlewareToBePiped)) {
                 $middlewareToBePiped = $serviceLocator->get($middlewareToBePiped);
             }
-            if (! $middlewareToBePiped instanceof MiddlewareInterface && ! is_callable($middlewareToBePiped)) {
+            if (! $middlewareToBePiped instanceof MiddlewareInterface
```

Comment:
We don't need this line anymore, please check line 11 in this file:
```php
use Interop\Http\ServerMiddleware\MiddlewareInterface;
```

it should be removed.

You are using MiddlewareInstance from `webimpress/http-middleware-compatibility` so it's enough.

Reply:
It is needed for stratigility 2.0.x

Reply:
@Xerkus ok, fine, but we are not testing it then... Maybe we should set requirements to ^2.0.1 as it was and write additional test?

Reply:
@Xerkus I've added tests in #262

Reply:
Not really _supporting_ it any more but allowing it to be used anyway as it is an optional dependency. If it was direct dependency, I would have bumped it to ^2.1.x and dropped that check.

Reply:
As we keep this condition here, it should be also tested, this is why I've added it in PR #262 :)